oshdb-util/src/test/resources/relations/invalid-inner-rings.osm

```diff
@@ -0,0 +1,32 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<osm version="0.6">
+
+  <relation id="1" version="1" timestamp="2014-01-01T00:00:00Z" uid="1" user="test" changeset="1">
+    <member type="way" ref="1" role="outer"/>
+    <member type="way" ref="2" role="inner"/>
+    <member type="way" ref="2" role="inner"/>
+    <member type="way" ref="2" role="inner"/>
```

Comment:
Why multiple times the same way as member?

Reply:
this is what triggered the bug in #334: the second identical inner way gets merged with the first one (in the handle-touching-inner rings routine), and both "collapse" into a single node. The third way was then finding the degenerate "ring" from before (because `remove` didn't actually remove all of the pointers to that ring, see the `removeAll` fix above), causing the exception later on.

https://www.openstreetmap.org/relation/8899/history had this (or a related) issue in one of its early versions.